docs/porting/connectivity/Layer3IP.md

```diff
@@ -0,0 +1,131 @@
+
+If the device has an off-board network stack, a driver needs to implement [NetworkStack](../porting/networkstack.html) directly instead of passing network calls to the offboard stack.
+
+### Abstractions
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/l3ip-driver.png)<span>L3IP driver</span></span>
+
+The L3IP interface abstracts network stacks and drivers and easily permits multiple instances. The key API classes are:
+
+- `NetworkInterface` - an Mbed OS network interface of any type.
+- `NetworkStack` - an Mbed OS network stack of any type (may be off-board).
+- `OnboardNetworkStack` - an onboard network stack.
+- `L3IP` - an IP Layer3 cellular device driver.
+- `NetStackMemoryManager` - a memory manager used to pass data between driver and stack.
+- `L3IPInterface`- a `NetworkInterface` that uses an `L3IP` driver and an `OnboardNetworkStack`.
+
+### L3IP driver core
+
+The first step in the port is to create a driver class for the cellular device that you can instantiate to control your device. You must derive this class from the `L3IP` class. A network stack (or test framework) uses this API to control your driver.
+
+Class L3IP is entirely abstract - you need to implement about a dozen calls to activate the driver, send and receive packets and perform other control and information functions.
+
+There are also callback registration functions for upcalls from the driver - the stack can register callback functions for packet reception and link status changes.
+
+#### Initialization steps
+
+The L3IP driver class is instantiated during the creation of the network interface. When the network interface is brought up, the network stack powers the L3IP driver.
+
+Steps that the network stack uses to power the L3IP driver:
+
+1. The network stack configures the IP Layer3 memory manager class reference for the driver.
+1. The network stack sets the link input and state callbacks.
+1. The network stack calls the driver's power up function.
+1. The network stack reads the MTU size from the driver.
+
+**Optional steps:**
+
+1. The network stack might query the interface name from the driver.
+1. The network stack might configure multicast filtering.
+   1. The driver can either support multicast filtering or provide all frames.
+1. The network stack might query for the memory buffer align preference from the driver.
+   1. The network stack is not required to use the alignment for the memory buffers on link out.
+
+### IP Layer3 memory manager
+
+The IP Layer3 memory manager class provides an abstracted memory interface toward memory modules used in different network stacks. For the send and receive paths, data is transferred in memory buffers controlled through an `NetStackMemoryManager` object. The network stack, using an L3IP driver, provides it with a reference to the memory manager in use before powering up - this is constant as long as the L3IP is powered.
```

Comment:
This last sentence confuses me. Could you please clarify?

Reply:
It's based on EMAC  which has the same memory management arch and behavior. 
https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/porting/connectivity/ethernet.md
However to keep it more easy to understand than original i replaced  description to:
"The network stack provides L3IP driver  with reference to the memory manager before L3IP device powers up."